### PR TITLE
Adding projects to volunteer logs endpoints

### PR DESCRIPTION
**Pull request by eliascodes**
### Changes
* Set foreign keys in volunteer logs to `NULL` when project is marked as deleted
* Allow `POST`, `PUT` and `/sync` volunteer log endpoints to accept projects in the payload
* Add new resource name to be used for volunteer project related endpoints
   * Sorry for sneaking this into a PR instead of discussing first in an issue, but I hope my reasoning in 10bd52b makes sense. LMK if not.

### CODE REVIEW

**Comment by eliascodes** [comment]
@astroash did you rebuild your database before testing locally?

**Comment by astroash** [comment]
@eliascodes yup

**Comment by eliascodes** [comment]
> get /v1/users/volunteers/6 {} returns 401

This is because for some reason in the seeds user 6 has volunteer logs with both orgs 1 and 2. `1@aperture...` is admin of org 1, but user 6 is volunteer with org 2, so `requireSiblingUser` fails with a `401` (which should actually be a `403`)

I can remove this stray log for org 1, and this will go away (but other tests might break)

> get /v1/community-businesses/me/volunteers/projects returns 403

I don't see this for either user. I don't think your DB is reset.

**Comment by astroash** [comment]
project_woes

**Comment by astroash** [comment]
project_woes

note my dev and test db are the same

**Comment by eliascodes** [comment]
In your GIF you're on the staging branch.

**Comment by astroash** [comment]
Was responding to your comment re my db not being reset. Will check on this branch when I get a chance.

**Comment by astroash** [comment]
removed my last comment as i was being an idiot, soz 🤦🏽‍♀️

File: src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ describe('API /community-businesses/me/volunteers/projects', () => {
         method: 'GET',
         url: '/v1/community-businesses/me/volunteers/projects',
         credentials: {
-          scope: ['organisation_details-parent:read'],
+          scope: ['organisations_volunteer-parent:read'],
```
Comment (astroash) [inline src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts]: `organisations_volunteer` -> `organisations_volunteers`

all other permissions are plural.

Can you also add organisations_volunteers to `docs/permissions.md`

**Comment by astroash** [comment]
@eliascodes, can these routes also be updated in `api.json`

**Pull request merged**